pq/pq.go

```diff
@@ -71,53 +82,47 @@ func MakeRoot() [SzRoot]byte {
 // start transactions. An error is returned if the delegate is nil, the queue
 // header is invalid, some settings are invalid, or if some IO error occurred.
 func New(delegate Delegate, settings Settings) (*Queue, error) {
+	const op = "pq/new"
+
 	if delegate == nil {
-		return nil, errNODelegate
+		return nil, errOp(op).of(InvalidParam).report("delegate must not be nil")
 	}
 
-	accessor, err := makeAccess(delegate)
-	if err != nil {
-		return nil, err
+	accessor, k := makeAccess(delegate)
```

Comment:
I would write `errKind` instead of `k`. It's not obvious what's the second variable `makeAccess` return.